lib/chef/provider/package/chocolatey.rb

```diff
@@ -211,11 +211,18 @@ def args_to_string(*args)
         def available_packages
           @available_packages ||=
             begin
+              name = 0
+              version = 1
+
               cmd = [ "list -ar #{package_name_array.join ' '}" ]
               cmd.push( "-source #{new_resource.source}" ) if new_resource.source
-              parse_list_output(*cmd).reject do |name,version|
-                desired_name_versions[name] && desired_name_versions[name] != version
+              available = {}
+              parse_list_output(*cmd).each do |name_version|
```

Comment:
i'd probably just do something like:

```ruby
parse_list_output(*cmd).each do |name_version|
  name, version = name_version
```

then `name` and `version` are really the name and version in the loop